File: pytools/data/data.code.java.client/AckRoleLoginOk.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class AckRoleLoginOk
{
	private String uname;


	public AckRoleLoginOk(Packet packet)
	{
		uname = packet.readString();
	}



	public String getUname()
	{
		return this.uname;
	}

}

File: pytools/examples/php/client/protocol/AckRoleLoginOk.php
<?php
namespace protocol;

use network\Packet;


class AckRoleLoginOk
{
	private $uname = '';


	public function __construct(&$packet)
	{
		$this->decode($packet);
	}

	public function decode(&$packet)
	{
		$this->uname = $packet->readString();
	}


	public function getUname()
	{
		return $this->uname;
	}

}
